### ISSUE TITLE
[DECOHack] A_WeaponJump doesn't accept 256 as a valid chance parameter

### ISSUE THREAD

**Issue by Altazimuth** [MTrop/DoomTools#62]
Given the DECOHack
```
#include "classpath:decohack/mbf21.dh"

state S_PISTOL1
{
    PISG A 4 A_WeaponJump(S_SGUN1, 256)
}
```
when compiling you'll get the error `Invalid value '256' for WeaponJump arg 1: value must be between 0 and 255`. This is incorrect, as 256 is the maximum (sensible) value.

The comparison in dsda-doom is `if (P_Random(pr_mbf21) < psp->state->args[1])`, which means that if P_Random returns 255 then if the arg is 255 it _won't_ jump. Due to this a chance of 256 is required for it to _always_ jump.

It may be worth checking any other chance-related codepointers to see if they have similar behaviour and require tweaking in DECOHack.

**Comment by MTrop** [comment]
A new type will either have to be created for a strict check ("probability byte" (0-256)) or it can be changed to a `uint` without much need to keep in the byte range.

DSDA-Doom should be used as the reference implementation, in any case.

**Comment by MTrop** [comment]
Fixed in commit 9580ef9
